### ISSUE TITLE
Error while reading laz data

### ISSUE THREAD

**Issue by zarov** [visgl/loaders.gl#865]
I have some trouble reading some `laz` data, using `@loaders.gl/las` without the other package. While I succeed to parse my file correctly by using `laz-perf` directly, I get this error if I use `LASLoader.parse(data, options)`:
```
Uncaught (in promise) Error: Failed to read data: undefined
    at LAZLoader.readData (laslaz-decoder.js?5487:243)
    at LASFile.readData (laslaz-decoder.js?5487:350)
    at parseLASChunked (parse-las.js?5799:116)
    at parseLAS (parse-las.js?5799:18)
    at _callee$ (las-loader.js?80b7:34)
    at tryCatch (runtime.js?96cf:45)
    at Generator.invoke [as _invoke] (runtime.js?96cf:274)
    at Generator.prototype.<computed> [as next] (runtime.js?96cf:97)
    at asyncGeneratorStep (asyncToGenerator.js?1da1:3)
    at _next (asyncToGenerator.js?1da1:25)
    at eval (asyncToGenerator.js?1da1:32)
    at eval (asyncToGenerator.js?1da1:21)
    at parse (las-loader.js?80b7:45)
```

It seems to fail on the first [`instance.getPoint`](https://github.com/visgl/loaders.gl/blob/5df6102f4c30a9984f9dd92a4c53aedae84deb08/modules/las/src/lib/laslaz-decoder.js#L227), but I can't understand how, as everything beforehand is correct.

Sidenote: I cloned this repo to debug it a little more, but I can't run things like `yarn test`; I get a lot of `Cannot find module '@loaders.gl/{module}' or its corresponding type declarations.` errors, and I don't know how to solve this.

**Comment by ibgreen** [comment]
>  I cloned this repo to debug it a little more, but I can't run things like yarn test; 

The first thing you need to do is `yarn bootstrap` to install packages, pre-build workers etc. Let me know if that helps

**Comment by ibgreen** [comment]
> While I succeed to parse my file correctly by using laz-perf directly, I get this error if I use LASLoader.parse(data, options):

With laz-perf I assume you are referring to the laz-perf module on npm, not `las-perf.js` inside loaders.gl.

What data are you passing in? An ArrayBuffer?

While you are certainly welcome to use `LAZLoader.parse` directly, all loaders.gl loaders are designed to be used together with loaders.gl/core `load` or `parse` function, they allow you to supply a wider variety of parameters etc.

**Comment by zarov** [comment]
> The first thing you need to do is `yarn bootstrap` to install packages, pre-build workers etc. Let me know if that helps

It is all working well now, thanks ! Maybe it can be added in `CONTRIBUTING` ;)



> With laz-perf I assume you are referring to the laz-perf module on npm, not `las-perf.js` inside loaders.gl.

I am talking the one from **hobu**, which is a little more up-to-date than the npm one, but yes. However, isn't [this one](https://github.com/visgl/loaders.gl/blob/master/modules/las/src/libs/laz-perf.js) the same ?

> What data are you passing in? An ArrayBuffer?

Yes, as it seems to be done in loaders.gl too if I'm not mistaken.

> While you are certainly welcome to use `LAZLoader.parse` directly, all loaders.gl loaders are designed to be used together with loaders.gl/core `load` or `parse` function, they allow you to supply a wider variety of parameters etc.

Yes, I'll maybe go to that if I can't make it work otherwise !

I'll close this issue in a few days to see if I can debug my problem, if it's okay with you.

**Comment by ibgreen** [comment]
> However, isn't this one the same ?

Yes, should be

> An ArrayBuffer, as it seems to be done in loaders.gl too if I'm not mistaken.

Yes

> I'll close this issue in a few days to see if I can debug my problem, if it's okay with you.

Of course. I am keen to understand what problem you are running into. loaders.gl is designed to be "plug-and-play"...

**Comment by zarov** [comment]
So I checked to load another file, and it worked without any problem ! But still, the initial file is correct when running a `laszip -check file.laz`.

My first thought was that there was a problem with the size of my file, being relatively small (less than 10k points), and maybe this was a problem when trying to read a chunk too big ([see here](https://github.com/visgl/loaders.gl/blob/master/modules/las/src/lib/parse-las.js#L117), this is bigger than the size of my file), but this wasn't the problem. Also, the header is correctly read before that, so the file seems really fine.

After logging a lot of things and finding nothing, I tried to debug `laz-perf` but I couldn't find some methods, like the function `getPoint` that is the whole problem. Can you explain where can I find this part ? Thanks ! I suspect this has something to do with what people added to this file, compared to the hobu version.

**Comment by ibgreen** [comment]
This code hasn't been touched in a while, but the laz-perf itself should not be modified. I believe that any additions to this file would be very minimal and were just put there to help package this as a loaders.gl loader object. 

It is certainly possible that we are using an older version of laz-perf, and that it is time to update it. You could try to replace the one in loaders.gl with the one from the laz-perf module and see if that helps. If it does we can update the las module.

I assume you are not able to share any of the files that don't load?

**Comment by zarov** [comment]
Compiling a new version from the cpp repo and replacing it here makes everything works ! It seems that the version used here is very old (circa 2014), while the last updates in laz-perf were made last year. The culprit may be `extra bytes`, which support has been made in 2018 (see this https://github.com/hobu/laz-perf/pull/56).

I'm looking into sharing at least a file for you to test and confirm it.

**Comment by ibgreen** [comment]
This sounds great. Would you like to open a PR with the change?  

You'd get your name on books as a contributor and be listed on the https://loaders.gl/ homepage.

**Comment by zarov** [comment]
Yes, I'll do that !

**Comment by zarov** [comment]
You can test the problem with this file: https://github.com/hobu/laz-perf/blob/master/test/raw-sets/extrabytes.laz

**Comment by zarov** [comment]
Solved with #869
